src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -15,6 +15,7 @@
 // </copyright>
 
 #nullable enable
+#pragma warning disable 0436
```

Comment:
We should remove this. Here is what is happening:

we have 2 projects:

`OpenTelemetry.Api` (targets netstandard2.0;net462)
`OpenTelemetry` (targets net6.0;netstandard2.1;netstandard2.0;net462) and references OpenTelemetry.Api

The `UnconditionalSuppressMessageAttribute` is defined in `net6.0`, but isn't defined in `netstandard2.0`.  When we build `OpenTelemetry.Api`, we are putting the type `UnconditionalSuppressMessageAttribute` as an internal type into `OpenTelemetery.Api`, since that attribute isn't defined in `netstandard2.0`.

Normally this is fine because only that assembly sees types that are internal. However, `OpenTelemetry` can see `OpenTelemetry.Api`'s internals because it has an InternalsVisibleTo relationship.

When `OpenTelemetry` builds for `net6.0`, it sees 2 different `UnconditionalSuppressMessageAttribute` types - one in the `OpenTelemetry.Api` `netstandard2.0` assembly (through internals visible to). and one in .NET itself. So the compiler complains.

The easiest fix for this is to change the TargetFrameworks property in `OpenTelemetry.Api.csproj` to be:

`<TargetFrameworks>net6.0;netstandard2.0;net462</TargetFrameworks>`

(add `net6.0`)

Then when `OpenTelemetry` builds for net6.0, it won't see the attribute from `OpenTelemetry.Api` anymore, since it won't be there. It will only see the attribute from the framework in `net6.0`